commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -603,3 +603,36 @@ def retire_customer_anonymize_fields(self, customer_id: str, customer_version: i
                          f"with ID: {customer_id}, after LMS retirement with "
                          f"error correlation id {err.correlation_id} and error/s: {err.errors}")
             raise err
+
+    def is_first_time_discount_eligible(self, email: str) -> bool:
+        """
+        Check if a user is eligible for a first time discount
+        Args:
+            email (str): Email of the user
+        Returns (bool): True if the user is eligible for a first time discount
+        """
+        try:
+            discounts = self.base_client.discount_codes.query(
+                where="code in :discountCodes",
+                predicate_var={'discountCodes': settings.COMMERCETOOLS_FIRST_TIME_DISCOUNTS}
+            )
+            discount_ids = [discount.id for discount in discounts.results]
+
+            discounted_orders = self.base_client.orders.query(
+                where=[
+                    "customerEmail=:email",
```

Comment:
We can maybe save two API calls from being made here.
First thought on my mind was to use reference expansion which I couldn't find any for discounts. I think you have also checked this.
The second option here is to store the discount IDs themselves in settings/edx-internal.
We are already following this pattern for line_item_transition: [Reference](https://github.com/edx/terraform/blob/899d2e9f51d70006ac82b9e7a4bdc437dfea3a80/plans/edx/apps/commerce-coordinator/prod.tfvars#L27C35-L27C71)
It is not normally suggested to store the database identifiers like this but as long as the discounts remain the same, the ids/keys are going to be the same so we can minimize the extra API call here which makes more sense to me.
@shafqatfarhan to weigh in on this.

Reply:
For the first point, we cannot query expanded objects. For the second option, I wanted the identifiers to be consistent across all environments, which is why I suggested using discount codes, as storing IDs didn’t seem ideal. However, if we're okay with this approach, we can certainly save an API call here